File: lib/utils/dock_utils.dart
part of dock_spawn;

int getPixels(String pixels) {
  if (pixels == null) return 0;
  return int.parse(pixels.replaceAll("px", ""));
}


Point2 getMousePosition(MouseEvent e, Element element) {
  int parentOffsetX = element.offsetLeft;
  int parentOffsetY = element.offsetTop;
  int parentWidth = element.clientWidth;
  int parentHeight = element.clientHeight;
  int x = e.offsetX - parentOffsetX;
  int y = e.offsetX - parentOffsetY;
  return new Point2(x, y);
}

void disableGlobalTextSelection() {
  document.body.classes.add("disable-selection");
}

void enableGlobalTextSelection() {
  document.body.classes.remove("disable-selection");
}

bool isPointInsideNode(int px, int py, DockNode node) {
  Element element = node.container.containerElement;
  int x = element.offsetLeft;
  int y = element.offsetTop;
  int width = element.clientWidth;
  int height = element.clientHeight;
  
  return (px >= x && px <= x + width && py >= y && py <= y + height);
}


class Rectangle {
  num x;
  num y;
  num width;
  num height;
}

File: js/lib/utils/DockUtils.js
function getPixels(pixels)
{
    if (pixels == null)
        return 0;
    return parseInt(pixels.replace("px", ""));
}

function disableGlobalTextSelection()
{
    document.body.classList.add("disable-selection");
}

function enableGlobalTextSelection()
{
    document.body.classList.remove("disable-selection");
}

function isPointInsideNode(px, py, node)
{
    var element = node.container.containerElement;
    var x = element.offsetLeft;
    var y = element.offsetTop;
    var width = element.clientWidth;
    var height = element.clientHeight;

    return (px >= x && px <= x + width && py >= y && py <= y + height);
}

function Rectangle()
{
//    num x;
//    num y;
//    num width;
//    num height;
}

function getNextId(prefix)
{
    return prefix + getNextId.counter++;
}
getNextId.counter = 0;

function removeNode(node)
{
    if (node.parentNode == null)
        return false;
    node.parentNode.removeChild(node);
    return true;
}

function Point(x, y)
{
    this.x = x;
    this.y = y;
}
